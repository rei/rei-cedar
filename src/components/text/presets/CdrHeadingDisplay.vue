<script setup lang="ts">
import { useCssModule, computed } from 'vue';
import { baseTextProps } from '../../../types/interfaces';

defineOptions({
  name: 'CdrHeadingDisplay',
});

interface headingDisplayTextProps extends baseTextProps {
   /** 
   * Sets the type scale
   * @type scaleValue
   * @values 2,3,4,5,6,7
   */
  scale?: '2'|'3'|'4'|'5'|'6'|'7',

}

const props = withDefaults(defineProps<headingDisplayTextProps>(), {
  tag: 'h1',
  scale: '7',
});

const typeProperties = computed(() => {
  return {
    '--cdr-heading-display-font-size': `var(--cdr-type-scale-${props.scale})`,
    '--cdr-heading-display-line-height': 
      `calc(var(--cdr-type-scale-${props.scale}) * var(--cdr-heading-display-line-height-ratio))`
  }
});

const baseClass = 'cdr-heading-display';
const style = useCssModule();
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
    :style="typeProperties"
  >
    <slot />
  </component>
</template>

<style module lang="scss">
@import "./styles/CdrPresets.module.scss";

.cdr-heading-display {
  @include cdr-text-base-mixin;
  @include cdr-text-heading-display-1600;
  font-size: var(--cdr-heading-display-font-size);
  line-height: var(--cdr-heading-display-line-height);
}
</style>  