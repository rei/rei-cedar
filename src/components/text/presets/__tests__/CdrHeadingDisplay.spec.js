import { mount } from '../../../../../test/vue-jest-style-workaround.js';
import CdrHeadingDisplay from '../CdrHeadingDisplay.vue';


describe('CdrHeadingDisplay', () => {
  describe('default configuration', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingDisplay, {
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  describe('passing a scale', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(CdrHeadingDisplay, {
        props: { scale: '4' },
        slots: {
          default: 'paragraph'
        }
      });
    });
    it('renders correctly', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });

});
