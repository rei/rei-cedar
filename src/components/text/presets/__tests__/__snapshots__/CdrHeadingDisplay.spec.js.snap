// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CdrHeadingDisplay > default configuration > renders correctly 1`] = `
<p
  class="cdr-text cdr-heading-display"
  style="--cdr-heading-display-font-size: var(--cdr-type-scale-7); --cdr-heading-display-line-height: calc(var(--cdr-type-scale-7) * var(--cdr-heading-display-line-height-ratio));"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;

exports[`CdrHeadingDisplay > passing a scale > renders correctly 1`] = `
<p
  class="cdr-text cdr-heading-display"
  style="--cdr-heading-display-font-size: var(--cdr-type-scale-4); --cdr-heading-display-line-height: calc(var(--cdr-type-scale-4) * var(--cdr-heading-display-line-height-ratio));"
>
  <!-- @slot CdrText content -->
  
  
  paragraph
  
  
</p>
`;
